lms/djangoapps/lti_provider/tests/test_views.py

```diff
@@ -170,11 +181,91 @@ def test_forbidden_if_session_incomplete(self):
                 'Expected Forbidden response when session is missing ' + key
             )
 
-    def test_session_cleared_in_view(self):
+    @patch('lti_provider.views.render_courseware')
+    def test_session_cleared_in_view(self, _render):
         """
         Verifies that the LTI parameters are cleaned out of the session after
         launching the view to prevent a launch being replayed.
         """
-        request = self.build_request()
+        request = build_run_request()
         views.lti_run(request)
         self.assertNotIn(views.LTI_SESSION_KEY, request.session)
+
+
+MODULE_INSTANCE = MagicMock()
+MODULE_INSTANCE.render = MagicMock(return_value='Fragment')
+
+
+@patch('lti_provider.views.render_to_response', return_value='Rendered page')
+@patch('lti_provider.views.get_module_by_usage_id', return_value=(MODULE_INSTANCE, None))
+@patch('lti_provider.views.has_access', return_value='StaffAccess')
+@patch('lti_provider.views.get_course_with_access', return_value='CourseWithAccess')
+@patch('lti_provider.views.CourseKey.from_string', return_value='CourseKey')
+class RenderCoursewareTest(TestCase):
+    """
+    Tests for the render_courseware method
+    """
+
+    def test_valid_launch(self, _key_mock, _course_mock, _access_mock, _module_mock, _render_mock):
```

Comment:
To avoid having all of these unused arguments passed to each test function, it might be better to define all these patches in a setUp method:
```python
def setUp(self):
    super(RenderCoursewareTest, self).setUp()
    p1 = patch(patch('lti_provider.views.render_to_response', return_value='Rendered page')
    p1.start()
    self.addCleanup(p1.stop)
    p2 = patch('lti_provider.views.get_module_by_usage_id', return_value=(MODULE_INSTANCE, None))
    p2.start()
    self.addCleanup(p2.stop)
    # etc...
```
If you want to be fancy, you could even define some kind of mapping of dotpath to return value and iterate over that, to avoid duplicated code in this setUp method. But that might make it more difficult to follow, so that's a judgement call. You should do whatever you think is clearer.